R/importSNV.TCGA.R

```diff
@@ -0,0 +1,323 @@
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,2] == "chr") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,3] == "start") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,4] == "end") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,5] == "strand") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,6] == "class") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,7] == "type") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,8] == "aRef") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,9] == "a1") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,10] == "a2") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,11] == "TCGA") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,12] == "UUID") {
+            headValid <- FALSE
+        }
+        return(headValid)
+    }
+
+    if (missing(rSNV)) {
+        rSNV <- "MAFtoSNV"
+
+        # add metdata
+        for (name in names(meta)) {
+            attr(rSNV, name) <- meta[[name]]
+        }
+
+        write("sym\tchr\tstart\tend\tstrand\tclass\ttype\taRef\ta1\ta2\tTCGA\tUUID",
+              rSNV, append=FALSE)
+
+        snvValidity <- TRUE
+    } else {
+        rSNVf <- read.table(rSNV, header = TRUE)[,1]
+        rowsBefore <- length(rSNVf)
+        myNotes <- c("Number of rows originally in rSNV - ", rowsBefore)
+        snvValidity <- rSNVheadercheck(rSNV)
+    }
+
+    # ==== READ DATA ===========================================================
+    filesFinished <- 0
+    numFilesInvalid <- 0
+    for (i in fMAF) {
```

Comment:
This is not a good use of a progress bar, because the number of files is going to be small. In this case I would produce feedback like:
```
Processing "myNewTCGAdata-12345.maf" ... done. (1/3)
```
etc.